### PR TITLE
Use matplotlib-inline

### PR DESCRIPTION
**Pull request by martinRenou**
This goes in the direction of #588

Opening as a draft for now, until there is some consensus on under which organization we put this matplotlib-inline (and how we should name it). 

It is temporarily here: https://github.com/martinRenou/matplotlib-inline

I believe this should go under http://github.com/matplotlib or http://github.com/ipython

### CODE REVIEW

File: ipykernel/pylab/backend_inline.py [hunk 1]
```diff
@@ -3,246 +3,4 @@
...
-    opposed to dark). Based on ITU BT.601 luminance formula (see
-    https://stackoverflow.com/a/596241)."""
-    rgbaArr = colors.to_rgba_array(color)
-    return rgbaArr[:,:3].dot((.299, .587, .114)) > .5
-
-def _is_transparent(color):
-    """Determine transparency from alpha."""
-    rgba = colors.to_rgba(color)
-    return rgba[3] < .5
+from matplotlib_inline.backend_inline import * # analysis: ignore
```
Comment (martinRenou) [inline ipykernel/pylab/backend_inline.py:6]: We might want to add a deprecation warning here.

**Comment by martinRenou** [comment]
This PR is coupled with https://github.com/ipython/ipython/pull/12817, but it doesn't need it for having a working inline backend.

**Comment by SylvainCorlay** [comment]
@martinRenou I rebased the PR now that IPython 7.23 is out with the new inline backend.

**Comment by martinRenou** [comment]
Thanks! I'll see if this PR needs updating

**Review by blink1073 — APPROVED** [review]
Thanks!

**Pull request merged**